pkg/task/v1alpha1/cstor_snapshot_delete.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2018 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cstor "github.com/openebs/maya/pkg/snapshot/cstor/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// cstorSnapshotDelete represents a cstor snapshot delete runtask command
+//
+// NOTE:
+//  This is an implementation of CommandRunner
+type cstorSnapshotDelete struct {
+	cmd *RunCommand
+}
+
+// Run deletes cstor snapshot contents
+func (c *cstorSnapshotDelete) Run() (r RunCommandResult) {
+	// api call to list snapshots and snapshot actions per controller
+	ip, _ := c.cmd.Data["ip"].(string)
```

Comment:
I guess read & delete has lot or redundant code.
Can this be moved to `cstorsnapshotcommand` struct

Reply:
Something like ```go
// validateOptions checks if the required params are missing
func (c *cstorSnapshotCommand) validateOptions() error {
	ip, _ := c.cmd.Data["ip"].(string)
	volName, _ := c.cmd.Data["volname"].(string)
	snapName, _ := c.cmd.Data["snapname"].(string)
	if len(ip) == 0 {
		return errors.Errorf("missing ip address")
	}

	if len(volName) == 0 {
		return errors.Errorf("missing volume name")
	}

	if len(snapName) == 0 {
		return errors.Errorf("missing snapshot name")
	}
}``` ?
How would I call this method from `func (c *cstorSnapshotDelete) Run() (r RunCommandResult) `?